src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,138 @@
+
+        String line;
+        StringBuilder sb = new StringBuilder();
+
+        while ((line = buffered.readLine()) != null){
+            if(line.contains("review/userId:")){
+                String[] parts = line.split("review/userId: ");
+                if(!users.containsKey(parts[1])) {
+                    users.put(parts[1], userNum);
+                    usersList.add("" + userNum);
+                    userNum++;
+                } else{
+                    usersList.add("" + users.get(parts[1]));
+                }
+            }else if(line.contains("product/productId:")){
+                reviews++;
+                String[] parts = line.split("product/productId: ");
+
+                if(!products.containsKey(parts[1])) {
+                    products.put(parts[1], productNum);
+                    productsList.add("" + productNum);
+                    productNum++;
+                } else{
+                    productsList.add("" + products.get(parts[1]));
+                }
+            }else if(line.contains("review/score:")){
+                String[] parts = line.split("review/score: ");
+                scoresList.add(parts[1]);
+            }
+        }
+
+        for (int counter = 0; counter < usersList.size(); counter++) {
+            pw.write(usersList.get(counter) + ',' + productsList.get(counter) + ',' + scoresList.get(counter) + '\n');
+            pw.println();
+        }
+
+        pw.close();
+
+        //pw.write(sb.toString());
+        //pw.close();
+    } catch(FileNotFoundException e) {
+        System.err.println("Caught FileNotFoundException: " + e.getMessage());
+    } catch(IOException e) {
+        System.err.println("Caught IOException: " + e.getMessage());
+    }
+}
+
+public int getTotalReviews(){
+    return reviews;
+}
+
+public int getTotalProducts(){
+    return products.size();
+}
+
+public int getTotalUsers(){
+    return users.size();
+}
+
+public <String> List<String> getRecommendationsForUser(String userID){
+    List<RecommendedItem> recommendations = null;
+
+
+    try{
+        DataModel model = new FileDataModel(new File("movies.csv"));
+        UserSimilarity similarity = new PearsonCorrelationSimilarity(model);
+        UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+        UserBasedRecommender recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+        recommendations = recommender.recommend(users.get(userID), 3);
+
+    } catch(IOException e) {
+        System.err.println("Caught IOException: " + e.getMessage());
+    } catch(TasteException e){
+        System.err.println("Caught TasteException: " + e.getMessage());
+    }
+
+    List<String> results = new ArrayList<String>();
+    for (RecommendedItem recommendation : recommendations) {
+        results.add((String)getProductID((int)recommendation.getItemID()));
+    }
+
+    return results;
+}
+
+public String getProductID(int value){
```

Comment:
In this function, you may use the `products.contains(key)` to avoid explicitly iterate over the `KeySet` :)